src/Agent/Agent.ts

```diff
@@ -0,0 +1,20 @@
+import { BaseChatModel } from '@langchain/core/language_models/chat_models'
+import { PromptTemplate } from '@langchain/core/prompts'
+
+import { AgentInput } from './types'
+
+abstract class Agent {
+  protected model: BaseChatModel
+  protected prompt: PromptTemplate
+  protected input: AgentInput
+
+  protected constructor(model: BaseChatModel, prompt: string, input: AgentInput) {
```

Comment:
Are you going to put prompt string to Agent? As I can understand the Agent can have a set of prompts

Reply:
Fixed, used for now prompts as in langchain docs:

ChatPromptTemplate.fromMessages([
  ["system", "You are very powerful assistant, but don't know current events"],
  ["human", "{input}"],
  new MessagesPlaceholder("agent_scratchpad"),
])